skimage/io/_plugins/pil_plugin.py

```diff
@@ -43,18 +43,12 @@ def imread(fname, dtype=None, img_num=None, **kwargs):
         if fname.lower().endswith(('.tiff', '.tif')):
             return tif_imread(fname, **kwargs)
 
-    im = Image.open(fname)
-    try:
-        # this will raise an IOError if the file is not readable
-        im.getdata()[0]
-    except IOError as e:
-        site = "http://pillow.readthedocs.org/en/latest/installation.html#external-libraries"
-        pillow_error_message = str(e)
-        error_message = ('Could not load "%s" \n'
-            'Reason: "%s"\n'
-            'Please see documentation at: %s') % (fname, pillow_error_message, site)
-        raise ValueError(error_message)
+    if isinstance(fname, string_types):
+        with open(fname, 'rb') as f:
+            im = Image.open(f)
+            return pil_to_ndarray(im, dtype=dtype, img_num=img_num)
     else:
+        im = Image.open(fname)
```

Comment:
Couldn't this branch still go unclosed? I think this needs a context manager to match the prior `if` block.